src/main/java/org/example/homework4/LeapYear.java

```diff
@@ -0,0 +1,16 @@
+package org.example.homework4;
+
+public class LeapYear {
+    public Boolean isLeapYear(int year) {
+        if (year >= 1 && year <= 9999) {
+            if (year % 4 == 0 && year % 100 != 0) {
+                System.out.println("The " + year + " year is leap!");
+                return true;
+            } else if (year % 100 == 0 && year % 400 == 0) {
```

Comment:
The condition `year % 100 == 0 && year % 400 == 0` is incorrect. According to the rules for leap years, if a year is divisible by 100 but not divisible by 400, it is not a leap year.